presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1181,6 +1181,15 @@ private PhysicalOperation visitScanFilterAndProject(
                     channel++;
                 }
             }
+            //TODO: This is a simple hack, it will be replaced when we add ability to push down sampling into connectors.
```

Comment:
Add a comment explaining that this works because SYSTEM sampling is done by the coordinator by dropping entire splits

Reply:
Also, the check should ensure the sampling is of type SYSTEM

Reply:
I have updated the changes. Please let me know if there is any other changes.